bitcoin/src/blockdata/constants.rs

```diff
@@ -271,12 +271,13 @@ mod test {
         // Compare strings because the spec specifically states how the chain hash must encode to hex.
         assert_eq!(got, want);
 
+        #[allow(unreachable_patterns)] // This is specifically trying to catch later added variants.
         match network {
             Network::Bitcoin => {},
             Network::Testnet => {},
             Network::Signet => {},
             Network::Regtest => {},
-            // Update ChainHash::using_genesis_block and chain_hash_genesis_block with new variants.
+            _ => panic!("Update ChainHash::using_genesis_block and chain_hash_genesis_block with new variants"),
```

Comment:
OMG, please no. Put the comment on top of `match` instead. Anyway, we need a macro for this so the comment won't be needed.

Reply:
I'm quite confident this `panic` will be compiled away even with no optimizations, and it's only in a test ... and I'm skeptical that a macro would be easier to read/maintain.

Reply:
The point is previously it was impossible to forget adding the branch now it's possible without any warning.

Single source of truth is super-important. Breaking it kicked my ass so hard so many times I don't want to ever do it again.

Reply:
Oh, I now realized what it actually does. I'm OK with this for now but still think macro is superior.

Reply:
Because this is test code, we will get a warning -- the test failing with a panic.

I'll take your "OK for now" as an ok to merge this -- we can fix it to use a macro after the fact.

Reply:
I intended to review this right after release but I guess it's fine. I will complain when I see something problematic later.